### PR TITLE
Unit test activity service

### LINKED ISSUE #92: Add unit tests for services
- [x] UserService
- [ ] ProductService
- [x] CategoryService
- [ ] CommentService
- [ ] FavoriteService
- [ ] FollowService
- [ ] CategoryFollowService

Each service should cover: happy path, edge cases, and authorization/ownership checks where applicable

### PR DESCRIPTION
**Pull request by ahmedXmuhamed**
## Summary by CodeRabbit

* **Tests**
  * Added a comprehensive Activity service test suite covering activity retrieval, listing user activities, creation behavior, mapping of optional fields, input validation, and repository interaction order.
  * Removed the placeholder unit test file.

* **Chores**
  * Updated AutoMapper-related test dependencies and removed the dependency-injection helper package reference.

### CODE REVIEW

**Comment by ahmedmmordi** [comment]
@ahmedXmuhamed Squash your commits

**Comment by ahmedmmordi** [comment]
@ahmedXmuhamed Add the remaining to #92 , and mark the finished. Also FYI, they ordered by highest priority, at least first few ones..

File: src/backend/ReUse.Tests/Services/ActivityServiceTests.cs [hunk 1]
```diff
@@ -0,0 +1,392 @@
...
+        _activityRepo
+            .Setup(r => r.GetByUserIdAsync(userId, 50))
+            .ReturnsAsync(activities);
+
+        var result = await _sut.GetUserActivitiesAsync(userId);
+
+        Assert.NotNull(result);
+        Assert.Equal(3, result.Count);
+        Assert.All(result, dto => Assert.Equal(userId, dto.UserId));
+
```
Comment (ahmedmmordi) [inline src/backend/ReUse.Tests/Services/ActivityServiceTests.cs:184]: @ahmedXmuhamed Assert `TYPE` also

File: src/backend/ReUse.Tests/Services/ActivityServiceTests.cs [hunk 2]
```diff
@@ -0,0 +1,392 @@
...
+    }
+
+    // GUARD — limit of zero throws BadRequestException
+    [Fact]
+    public async Task GetUserActivitiesAsync_ThrowsBadRequest_WhenLimitIsZero()
+    {
+        var userId = Guid.NewGuid();
+
+        await Assert.ThrowsAsync<BadRequestException>(
+            () => _sut.GetUserActivitiesAsync(userId, 0));
```
Comment (ahmedmmordi) [inline src/backend/ReUse.Tests/Services/ActivityServiceTests.cs:238]: @ahmedXmuhamed Verify if the repo called ever or not like the above..

File: src/backend/ReUse.Tests/Services/ActivityServiceTests.cs [hunk 3]
```diff
@@ -0,0 +1,392 @@
...
+    }
+
+    // GUARD — negative limit throws BadRequestException
+    [Fact]
+    public async Task GetUserActivitiesAsync_ThrowsBadRequest_WhenLimitIsNegative()
+    {
+        var userId = Guid.NewGuid();
+
+        await Assert.ThrowsAsync<BadRequestException>(
+            () => _sut.GetUserActivitiesAsync(userId, -1));
```
Comment (ahmedmmordi) [inline src/backend/ReUse.Tests/Services/ActivityServiceTests.cs:251]: @ahmedXmuhamed Same

File: src/backend/ReUse.Tests/Services/ActivityServiceTests.cs [hunk 4]
```diff
@@ -0,0 +1,392 @@
...
+    }
+
+    // GUARD — limit over 1000 throws BadRequestException
+    [Fact]
+    public async Task GetUserActivitiesAsync_ThrowsBadRequest_WhenLimitExceeds1000()
+    {
+        var userId = Guid.NewGuid();
+
+        await Assert.ThrowsAsync<BadRequestException>(
+            () => _sut.GetUserActivitiesAsync(userId, 1001));
```
Comment (ahmedmmordi) [inline src/backend/ReUse.Tests/Services/ActivityServiceTests.cs:264]: @ahmedXmuhamed Same

**Review by ahmedmmordi — CHANGES_REQUESTED** [review]

**Review by Farahhazemm — APPROVED** [review]

**Comment by Farahhazemm** [comment]
@ahmedmmordi  Approve this & merge

**Review by ahmedmmordi — APPROVED** [review]
